internal/credential/vault/repository_credential_store.go

```diff
@@ -737,6 +736,15 @@ func (r *Repository) UpdateCredentialStore(ctx context.Context, cs *CredentialSt
 				msgs = append(msgs, returnedClientCert.oplogMessage(db.CreateOp))
 			}
 
+			ir, err := NewRepository(reader, w, r.kms)
```

Comment:
Why create a new repository when this call is in a transaction?

Reply:
Fixed this to instead use the same mechanism that Lookup uses inlined in the update call.